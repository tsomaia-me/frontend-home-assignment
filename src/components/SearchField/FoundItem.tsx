import React, { useCallback } from 'react'

export interface FoundItemProps {
  packageName: string
  onClick: (item: string) => void
}

function FoundItem({ packageName, onClick }: FoundItemProps) {
  const handleClick = useCallback(() => onClick(packageName), [packageName, onClick])

  return (
    <li onMouseDown={handleItemMouseDown} onClick={handleClick}>
      {packageName}
    </li>
  )
}

function handleItemMouseDown(event: React.MouseEvent) {
  event.preventDefault()
}

export default FoundItem
