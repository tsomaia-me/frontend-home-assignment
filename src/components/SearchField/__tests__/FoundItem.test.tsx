import React from 'react'
import renderer from 'react-test-renderer'
import FoundItem from '../FoundItem'

test('renders FoundItem correctly', () => {
  const json = renderer.create(
    <FoundItem
      packageName="react"
      onClick={console.log}
    />
  ).toJSON()
  expect(json).toMatchSnapshot()
})
