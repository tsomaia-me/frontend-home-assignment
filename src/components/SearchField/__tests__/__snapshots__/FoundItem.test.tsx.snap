// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FoundItem correctly 1`] = `
<li
  onClick={[Function]}
  onMouseDown={[Function]}
>
  react
</li>
`;
